api/src/main/java/com/codeforcommunity/api/ITeamsProcessor.java

```diff
@@ -5,13 +5,21 @@
 import com.codeforcommunity.dto.team.GetAllTeamsResponse;
 import com.codeforcommunity.dto.team.GetUserTeamsResponse;
 import com.codeforcommunity.dto.team.InviteMembersRequest;
+import com.codeforcommunity.dto.team.TeamApplicant;
 import com.codeforcommunity.dto.team.TeamResponse;
+import java.util.List;
 
 public interface ITeamsProcessor {
 
   TeamResponse createTeam(JWTData userData, CreateTeamRequest teamRequest);
 
-  void joinTeam(JWTData userData, int teamId);
+  void applyForTeam(JWTData userData, int teamId);
```

Comment:
"apply" seems weird given that it's 0% skill based for allowing who joins your team, but kinda nitpicky idk